### PR TITLE
PHPLIB-1162: Remove unused argument Explainable::getCommandDocument(Server $server)

### PR DESCRIPTION
**Pull request by GromNaN**
Fix PHPLIB-1162

### CODE REVIEW

File: src/Operation/Explainable.php [hunk 1]
```diff
@@ -32,5 +32,5 @@ interface Explainable extends Executable
      *
      * @return array
      */
-    public function getCommandDocument(Server $server);
```
Comment (GromNaN) [inline src/Operation/Explainable.php:35]: This interface is `@internal`, we can change it without breaking use code.

**Review by alcaeus — COMMENTED** [review]
Code changes LGTM. I'm a little bit on the fence about whether we should make this change.

FWIW, the `$server` parameter was added as back then the `FindAndModify` operation used it in its `createCommandDocument` method to appropriately add the `bypassDocumentValidation` option. However, this was removed as part of #893 when we cleaned up pre-3.6 logic. There could be future cases in which the command document might depend on a selected server to add the correct options, in which case we'd be adding the parameter back to every `Explainable`.

**Comment by GromNaN** [comment]
Interesting. To reduce coupling, it might be replaced by the server info array when the needs to check server version pops again.

**Comment by jmikola** [comment]
> Code changes LGTM. I'm a little bit on the fence about whether we should make this change.

@alcaeus: Thanks for saving me the trouble of digging up the historical references.

> To reduce coupling, it might be replaced by the server info array when the needs to check server version pops again.

The dominate use case for Server arguments in `execute()` methods (similar to this) is calling `server_supports_feature()`, which does only need `Server::getInfo()`. I don't feel strongly about this, though, since it would just put the onus on each operation to extract the min/max wire versions.

I reckon the only real benefit is we might be able to write unit tests for `server_supports_feature()` (IIRC mocking the server isn't feasible).

**Pull request merged**